### ISSUE TITLE
Неверное название слоя, при добавлении Garmin GDB

### ISSUE THREAD

**Issue by yellow-sky** [nextgis/NextGIS_QGIS_open#20]
Ошибка #9
При добавлении слоя gdb название слоя все так же 'waypoints', а не 'Название файла (waypoints)'
Такое поведение проявляется, если в слое только один подслой, и QGIS не выводит окно выбора подслоев, а сразу добавляет его на карту.

**Comment by alexbruy** [comment]
Пример файла можно?

**Comment by yellow-sky** [comment]
https://drive.google.com/open?id=0B8fD-l_uK4vFeTJjaXhmSWdocXFJQlFBSXgyMUdEZlNielRv&authuser=0
В архиве несколько гарминовских и gpx файлов

**Comment by alexbruy** [comment]
Судя по представленными примерам, файлы GARMIN GDB не являются составными. Соответственно имя и не должно иметь вид «layer (sublayer)»

**Comment by simgislab** [comment]
и gpx-ов в архиве вроде нет

**Comment by simgislab** [comment]
вру, есть

@alexbruy, а приложенный GPX? Или его слои нормально открывает?

![](http://m-d.me/img/ss/20150326-809-28kb.jpg)

**Comment by yellow-sky** [comment]
@alexbruy, вот пример многослойного garmina: http://www.historic66.com/gps/garmin/jim-mccool.zip
А еще, мы смотрим документацию: http://www.gpsbabel.org/htmldoc-development/fmt_gdb.html

**Comment by yellow-sky** [comment]
Обновил описание ошибки

**Comment by alexbruy** [comment]
Не воспроизводится с составными файлами Garmin GDB

**Comment by yellow-sky** [comment]
Саш, прочитай пожалуйста #9. 
Суть в том, что реализация #9 половинчатая - почему то для geojson теперь все классно и название прописывается верно, а вот для gdb нет.
Сам формат GDB - составной. В некоторых случаях, в файле присутствует только один подслой - например точечный (waypoints). И в это случае, после добавления десятка файлов, получается полная х**ня:
![image](https://cloud.githubusercontent.com/assets/1277422/6881442/4ab34abc-d571-11e4-9d36-17dd15f5ec47.png)
Это баг. Пользоваться этим невозможно. Если ты найдешь долб***а который скажет что так и должно работать - я поаплодирую тебе. Наши клиенты справедливо считают, что программа не должна так работать. 
Если ты так не считаешь, не закрывай пожалуйста этот тикет. Будем искать того кто сможет исправить такое поведение.

**Comment by simgislab** [comment]
Так, давайте не будем ругаться, Саша, не закрывай плиз так быстро тикеты. Дай третьей стороне проверить.

**Comment by simgislab** [comment]
GPX работают нормально.
![](http://m-d.me/img/ss/20155628-175657.png)

**Comment by simgislab** [comment]
@alexbruy похоже у меня то же, что и у @yellow-sky 
Первая - однослойный GDB, второе - многослойный

![](http://m-d.me/img/ss/20153128-183103.png)

**Comment by simgislab** [comment]
теперь даже шейп-файлы где один тип геометрии так показывает, что видимо не нормально

![](http://m-d.me/img/ss/20153008-173052.png)
